examples/moving_least_squares/moving_least_squares.cpp

```diff
@@ -0,0 +1,255 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+#include <ArborX_InterpMovingLeastSquares.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <boost/program_options.hpp>
+
+#include <fstream>
+#include <iostream>
+#include <optional>
+#include <random>
+
+using Point = ArborX::ExperimentalHyperGeometry::Point<2, double>;
+constexpr int DIM = ArborX::GeometryTraits::dimension<Point>::value;
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = typename ExecutionSpace::memory_space;
+
+// Randomly fills a 2 * half_edge box centered at 0
+void filledBoxRandom(double half_edge,
+                     Kokkos::View<Point *, MemorySpace> points)
+{
+  int n = points.extent(0);
+  auto points_host = Kokkos::create_mirror_view(points);
+
+  std::uniform_real_distribution<double> dist(-half_edge, half_edge);
+  std::random_device rd;
+  std::default_random_engine gen(rd());
+  auto random = [&dist, &gen]() { return dist(gen); };
+  for (int i = 0; i < n; ++i)
+    for (int d = 0; d < DIM; ++d)
+      points_host(i)[d] = random();
+
+  Kokkos::deep_copy(points, points_host);
+}
+
+// Evenly fills a 2 * half_edge box centered at 0
+void filledBoxEven(double half_edge, std::array<int, DIM> const line_len,
+                   Kokkos::View<Point *, MemorySpace> points)
+{
+  int n = points.extent(0);
+  auto points_host = Kokkos::create_mirror_view(points);
+
+  for (int i = 0; i < n; ++i)
+  {
+    int j = i;
+    for (int d = 0; d < DIM; ++d)
+    {
+      points_host(i)[d] =
+          2 * (j % line_len[d]) * half_edge / (line_len[d] - 1) - half_edge;
+      j = j / line_len[d];
+    }
+  }
+
+  Kokkos::deep_copy(points, points_host);
+}
+
+// Switches the set of points to the new base
```

Comment:
By default, everything is generated in a box. This lets skew and rotate the box however the user wants to. Like the previous comment, it can be removed, as well as the creation of the basis matrix.